packages/react-ui/src/app/builder/builder-hooks.ts

```diff
@@ -223,3 +224,67 @@ function getStateAtPath({
   });
   return targetMap;
 }
+
+const getAllStepsMentions: (state: BuilderState) => MentionTreeNode[] = (
+  state,
+) => {
+  const { selectedStep, flowVersion } = state;
+  if (!selectedStep || !flowVersion || !flowVersion.trigger) {
+    return [];
+  }
+  const step = flowHelper.getStep(flowVersion, selectedStep.stepName);
+  if (!step) {
+    return [];
+  }
+  const path = flowHelper.findPathToStep({
+    stepToFind: step,
+    trigger: flowVersion?.trigger,
+  });
+
+  return path.map((s) => {
+    const stepMentionNode: MentionTreeNode =
```

Comment:
Please use isNil in these cases isNils.settings.inputUiInfo?.currentSelectedData )